### PR TITLE
Course choice refactorings

### PR DESCRIPTION
**Pull request by malcolmbaig**
## Context

A series of small refactors identified while working on Trello card 883. 

## Changes proposed in this pull request

Code changes with the goal of improving readability.

## Guidance to review

See individual commits.

## Link to Trello card

https://trello.com/c/TU8oMtpf

## Things to check

- [x] This code doesn't rely on migrations in the same Pull Request
- [x] If this code includes a migration adding or changing columns, it also backfills existing records for consistency
- [x] API release notes have been updated if necessary
- [x] New environment variables have been [added to the Azure config](https://github.com/DFE-Digital/apply-for-postgraduate-teacher-training#azure-hosting-devops-pipeline)

### CODE REVIEW

File: app/helpers/select_options_helper.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+module SelectOptionsHelper
```
Comment (duncanjbrown) [inline app/helpers/select_options_helper.rb]: I can't see this included anywhere - is there some magic?

File: app/models/candidate_interface/pick_provider_form.rb [hunk 2]
```diff
@@ -5,8 +5,8 @@ class PickProviderForm
     attr_accessor :provider_id
     validates :provider_id, presence: true
 
-    def other?
-      Course.exposed_in_find.where(provider_id: provider_id).blank?
+    def courses_available?
```
Comment (duncanjbrown) [inline app/models/candidate_interface/pick_provider_form.rb]: This is so much better

File: app/helpers/select_options_helper.rb (hunk 1 above)
Comment (tvararu) [reply app/helpers/select_options_helper.rb]: I think Rails autoloads everything in `app/helpers`, but I may be mistaken!

Comment (malcolmbaig) [reply app/helpers/select_options_helper.rb]: > I can't see this included anywhere - is there some magic?

Yes - Rails makes all helpers available in all views.

File: app/controllers/candidate_interface/course_choices_controller.rb [hunk 3]
```diff
@@ -18,10 +18,9 @@ def have_you_chosen
 
     def make_choice
       @choice_form = CandidateInterface::CourseChosenForm.new(application_choice_params)
+      render(:have_you_chosen) && return if !@choice_form.valid?
```
[the remark this answers is not in the document]
Comment (malcolmbaig) [reply app/controllers/candidate_interface/course_choices_controller.rb]: Strong preference for `and return` but the linter complains with its current config. Making some changes to this based on an offline chat with @tijmenb - will revisit.

[the remark this answers is not in the document]
Comment (malcolmbaig) [reply app/controllers/candidate_interface/course_choices_controller.rb]: Updated to use `and`, with an accompanying commit that changes the rubocop config.

**Pull request merged**